fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabase.java

```diff
@@ -766,14 +766,23 @@ private Transaction createTransaction(@Nonnull FDBRecordContextConfig config,
         return transaction;
     }
 
+    /**
+     * Create an {@link FDBDatabaseRunner} for use against this database.
+     * @param contextConfigBuilder options for contexts opened by the new runner
+     * @return a new runner
+     */
+    @Nonnull
+    public FDBDatabaseRunner newRunner(@Nonnull FDBRecordContextConfig.Builder contextConfigBuilder) {
+        return new FDBDatabaseRunnerImpl(this, contextConfigBuilder);
+    }
 
     /**
      * Create an {@link FDBDatabaseRunner} for use against this database.
      * @return a new runner
      */
     @Nonnull
     public FDBDatabaseRunner newRunner() {
-        return new FDBDatabaseRunnerImpl(this);
+        return newRunner(FDBRecordContextConfig.newBuilder().setTransactionTimeoutMillis(factory.getTransactionTimeoutMillis()));
```

Comment:
It is a little weird that this sets the default option here... I suppose another approach would be to make the argument to `FDBDatabaseRunnerImpl` `@Nullable` and then fill it in with a default builder (like this) if `null` is provided, and then filling in the defaults is the job of in `FDBDatabaseRunnerImpl` (or there's a "default config builder" method in `FDBDatabase` and then _that_ does the job of filling in the default values).

I think all of those things could be changed later, though, without affecting the API.

Reply:
This turns out not to be necessary. The old code has some redundancy that was copied into the new. The default is to get from the factory where the default is to leave it alone from the FDB client's setting. There is no need to read from the factory here.